src/Omega_h_gmsh.cpp

```diff
@@ -581,6 +732,49 @@ void write(filesystem::path const& filepath, Mesh* mesh) {
   gmsh::write(stream, mesh);
 }
 
+
+#ifdef OMEGA_H_USE_GMSH
+
+void write_parallel(filesystem::path const& filename, Mesh& mesh) {
+  const auto dim = mesh.dim();
+  const auto nnodes = mesh.nverts();
+  const auto globals_v = mesh.globals(VERT);
+  const auto coords = mesh.coords();
+
+  std::vector<std::size_t> node_tags(static_cast<size_t>(nnodes));
+  std::vector<double> node_coords(3*static_cast<size_t>(nnodes), 0);
```

Comment:
Can you please confirm @smelchio  that the hard-coded 3 is normal because gmsh is always writing nodes in 3d?
If so I would replace all occurences of 3 by something a bit more explicit, for instance:

`const auto gmsh_dims = 3;`

Reply:
I confirm that gmsh always have 3 coordinates even for a 1D or a 2D mesh. If you are not in 3D, the z coordinate is not optional, i.e., coordinates are always a triplet (x, y , z) in gmsh (see e.g., http://gmsh.info/doc/texinfo/gmsh.html#File-formats )